examples/cpp/flow_control/SubscriberApp.cpp

```diff
@@ -119,36 +113,57 @@ void FlowControlExampleSubscriber::SubListener::on_subscription_matched(
     }
 }
 
-void FlowControlExampleSubscriber::SubListener::on_data_available(
+void SubscriberApp::on_data_available(
         DataReader* reader)
 {
     SampleInfo info;
-    FlowControlExample st;
-    if (reader->take_next_sample(&st, &info) == RETCODE_OK)
+    FlowControl st;
+    while ((!is_stopped()) && (RETCODE_OK == reader->take_next_sample(&st, &info)))
```

Comment:
Can I ask you to add the check for `!is_stopped()` to the xtypes subscriber `on_data_available`?